From 34a47af9dd66a09d15cd77f0d1bac969b3a8939f Mon Sep 17 00:00:00 2001
From: Ning Sun <ning.sun@intel.com>
Date: Wed, 24 Feb 2016 15:21:58 -0800
Subject: Don't skip first argument in Linux kernel command line

Since 429:664e696da669, tboot doesn't skip the first argument of a
given command line any more. Condequently, it shoudldn't be skipped
either when passed to the kernel in expand_linux_image().

Meanwhile, as skip_filename() isn't used any more, remove the code.

Signed-off-by Martin Wilck <martin.wilck@ts.fujitsu.com>
Signed-off-by Ning Sun <ning.sun@intel.com>

As skip_filename isn't used any more, remove the code.
---
 tboot/common/cmdline.c  |   17 -----------------
 tboot/common/linux.c    |    2 +-
 tboot/include/cmdline.h |    1 -
 3 files changed, 1 insertions(+), 19 deletions(-)

diff --git a/tboot/common/cmdline.c b/tboot/common/cmdline.c
index 7a037fd..c5fbf50 100644
--- a/tboot/common/cmdline.c
+++ b/tboot/common/cmdline.c
@@ -586,23 +586,6 @@ bool get_linux_mem(uint64_t *max_mem)
     return true;
 }
 
-const char *skip_filename(const char *cmdline)
-{
-    if ( cmdline == NULL || *cmdline == '\0' )
-        return cmdline;
-
-    /* strip leading spaces, file name, then any spaces until the next
-     non-space char (e.g. "  /foo/bar   baz" -> "baz"; "/foo/bar" -> "")*/
-    while ( *cmdline != '\0' && isspace(*cmdline) )
-        cmdline++;
-    while ( *cmdline != '\0' && !isspace(*cmdline) )
-        cmdline++;
-    while ( *cmdline != '\0' && isspace(*cmdline) )
-        cmdline++;
-    return cmdline;
-}
-
-
 /*
  * Local variables:
  * mode: C
diff --git a/tboot/common/linux.c b/tboot/common/linux.c
index 4947e9b..85e75a6 100644
--- a/tboot/common/linux.c
+++ b/tboot/common/linux.c
@@ -302,7 +302,7 @@ bool expand_linux_image(const void *linux_image, size_t linux_size,
            (unsigned long)(real_mode_base + real_mode_size));
 
     /* copy cmdline */
-    const char *kernel_cmdline = skip_filename(get_cmdline(g_ldr_ctx));
+    const char *kernel_cmdline = get_cmdline(g_ldr_ctx);
 
     printk(TBOOT_INFO"Linux cmdline placed in header: ");
     printk_long(kernel_cmdline);
diff --git a/tboot/include/cmdline.h b/tboot/include/cmdline.h
index a10e405..3259ea1 100644
--- a/tboot/include/cmdline.h
+++ b/tboot/include/cmdline.h
@@ -60,7 +60,6 @@ extern void linux_parse_cmdline(const char *cmdline);
 extern bool get_linux_vga(int *vid_mode);
 extern bool get_linux_mem(uint64_t *initrd_max_mem);
 
-extern const char *skip_filename(const char *cmdline);
 extern uint8_t get_loglvl_prefix(char **pbuf, int *len);
 
 #endif    /* __CMDLINE_H__ */
-- 
1.7.1

